pkg/configuration/configuration_test.go

```diff
@@ -82,15 +87,32 @@ func TestLoadFromSecret(t *testing.T) {
 			},
 		}
 
-		cl := test.NewFakeClient(t)
-		err := cl.Create(context.TODO(), secret)
-		require.NoError(t, err)
+		cl := test.NewFakeClient(t, secret)
+
+		// when
+		config, err := configuration.LoadConfig(cl)
 
-		config := configuration.LoadConfig(cl)
+		// then
+		require.NoError(t, err)
 		assert.Equal(t, "test-domain", config.GetMailgunDomain())
 		assert.Equal(t, "test-api-key", config.GetMailgunAPIKey())
 		assert.Equal(t, "test-sender-email", config.GetMailgunSenderEmail())
 	})
+
+	t.Run("secret not found", func(t *testing.T) {
+		// given
+		restore := test.SetEnvVarAndRestore(t, "HOST_OPERATOR_SECRET_NAME", "test-secret")
+		defer restore()
+
+		cl := test.NewFakeClient(t)
+
+		// when
+		config, err := configuration.LoadConfig(cl)
+
+		// then
+		require.Error(t, err)
```

Comment:
maybe we should check the error message here as well?

Reply:
added 1a7b7de